### PR TITLE
group: access group in critical section

### PR DESCRIPTION
**Pull request by XuNeo**
## Summary

~Make sure `tcb` still exist when we access members of it.~

As discussed below, function `group_argvstr`  should be renamed to `nxtask_argvstr` because it's no longer depends on group.

Also current caller to `group_argvstr` is procfs and assert handler. Both of them are now have `enter_critical_section ` protected, thus we don't need to add guard inside `group_argvstr` .

PR is updated accordingly.

## Impact

Should be minimal.

## Testing

Tested on internal projects that has frequent task create/destroy and a periodic call to procfs.

### CODE REVIEW

File: sched/group/group_argvstr.c [hunk 1]
```diff
@@ -60,16 +61,20 @@
...
   size_t n = 0;
 #ifdef CONFIG_ARCH_ADDRENV
   FAR struct addrenv_s *oldenv;
 #endif
 
+  flags = enter_critical_section();
+
   /* Sanity checks and idle tasks */
 
   if (!tcb || !tcb->group || !tcb->group->tg_info || size < 1 ||
```
Comment (anchao) [inline sched/group/group_argvstr.c:74]: Which statement in this function depends on group？ could we remove the group check and add critical section to caller？

Comment (XuNeo) [reply sched/group/group_argvstr.c:74]: Is it possible that after checking `!tcb->group`, CPU is switched out and `tcb->group->tg_info` triggers the issue?

Currently, only procfs and assert call this function.

Comment (XuNeo) [reply sched/group/group_argvstr.c:74]: I see. The direct access to tcb->group has been removed in [this commit](https://github.com/apache/nuttx/pull/12320/files#diff-4c80631b4969dc86c856de89c0a424189374e6284a10a8529662f9e15e3dccd0).

Looks like we do not need `tcb->group` any more. Should I drop the PR directly?

Comment (anchao) [reply sched/group/group_argvstr.c:74]: pleade remove the group check in this PR，which doesn't a necessary condition for this function

File: sched/group/group_argvstr.c [hunk 2]
```diff
@@ -60,16 +61,20 @@
 
 size_t group_argvstr(FAR struct tcb_s *tcb, FAR char *args, size_t size)
```
Comment (pussuw) [inline sched/group/group_argvstr.c:62]: Also the name group_argvstr is now obsolete, as the argument vector has been moved to TLS. Not a problem of this patch, but something worth noting that it needs clean-up.

Comment (XuNeo) [reply sched/group/group_argvstr.c:62]: Should we move `group/group_argvstr.c` to `task/task_argvstr.c` and rename it to `nxtask_argvstr`?

Comment (pussuw) [reply sched/group/group_argvstr.c:62]: That sounds good but I'm starting to wonder if https://github.com/apache/nuttx/pull/12320/files#diff-4c80631b4969dc86c856de89c0a424189374e6284a10a8529662f9e15e3dccd0 is wrong.

It removes the argument vector from group, which I think is wrong. Now pthreads do not inherit the process argument vector.

Comment (pussuw) [reply sched/group/group_argvstr.c:62]: Forget what I said, renaming to task_argvstr is just fine.

**Review by pussuw — APPROVED** [review]
LGTM!

**Pull request merged**